src/main/java/org/eclipse/cargotracker/interfaces/booking/facade/internal/DefaultBookingServiceFacade.java

```diff
@@ -28,6 +29,7 @@
 import org.eclipse.cargotracker.interfaces.booking.facade.internal.assembler.LocationDtoAssembler;
 
 @ApplicationScoped
+@Transactional
```

Comment:
I understand this is a way to avoid Hibernate lazy instantiation issues. However, this approach is a bit of a hack and counter to separation of concerns in DDD. There should be no reason for this to be transactional other than dealing with Hibernate. Instead, let us do a bit deeper domain analysis and see if lazy loading can be avoided altogether, probably using entity graphs. In the layers above the repository layer, you can introduce "load...For..." methods that are domain case specific. Ideally, this could extend to the repository layer as well. If not, you can introduce a finder parameter such as "LOAD_FULLY", "LOAD_PARTIALLY" or "LOAD_MINIMALLY" that maps to various entity graphs in the repository layer. This technique will keep things portable with Hibernate and still maintain proper layering instead of leaking data loading concerns into the business and UI tiers. Let me know if this all makes sense or we can discuss it in more detail.

Reply:
Currently, this is the simplest approach to avoid the lazy exception. As mentioned in #33 maybe define  @EntityGraph is the better solution. But for before experience, like what do in JBoss Seam, or Apache DeltaSpike(I do not remember the details), there is a generic **open session in view** like solution to apply to JSF request lifecycle.

Reply:
I really hate to say this, but I don't think we can have that approach in a DDD focused project. Do you think you could take some time to address this please? That would be awesome!